### PR TITLE
Refactor testing style for x/docs

### PR DESCRIPTION
**Pull request by LucaArietti**
## Checklist
- [ ] Targeted PR against correct branch
- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For Admin Use:
- [ ] Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- [ ] Reviewers Assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]"

### CODE REVIEW

File: x/docs/internal/keeper/keeper_test.go [hunk 1]
```diff
@@ -515,8 +684,59 @@ func TestKeeper_SaveDocumentReceipt_ExistingReceipt_DifferentUuid(t *testing.T)
...
+			} else {
+				expected := []types.DocumentReceipt{TestingDocumentReceipt}
+				require.Equal(t, expected, receipts)
+			}
+		})
+	}
 }
 
+/*
 func TestKeeper_UserReceivedReceiptsIterator_EmptyList(t *testing.T) {
```
Comment (gsora) [inline x/docs/internal/keeper/keeper_test.go:740]: Remove this old commented test if the new one has already been implemented.

File: x/docs/internal/keeper/keeper_test.go [hunk 2]
```diff
@@ -451,7 +520,107 @@ func TestKeeper_DocumentsIterator_ExistingList(t *testing.T) {
 // --- Document receipts
 // ----------------------------------
 
-func TestKeeper_SaveDocumentReceipt_EmptyList(t *testing.T) {
+func TestKeeper_SaveDocument(t *testing.T) {
+	tests := []struct {
+		name          string
+		empty         bool
```
Comment (gsora) [inline x/docs/internal/keeper/keeper_test.go:526]: What does "empty" mean? Empty receipts list?

File: x/docs/internal/keeper/keeper_test.go [hunk 3]
```diff
@@ -451,7 +520,107 @@ func TestKeeper_DocumentsIterator_ExistingList(t *testing.T) {
...
+		document      types.Document
+		receipt       types.DocumentReceipt
```
Comment (gsora) [inline x/docs/internal/keeper/keeper_test.go:528]: What's the difference between receipt and newReceipt?

File: x/docs/internal/keeper/keeper_test.go [hunk 4]
```diff
@@ -451,7 +520,107 @@ func TestKeeper_DocumentsIterator_ExistingList(t *testing.T) {
...
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cdc, ctx, k := SetupTestInput()
+
+			if tt.empty {
+				store := ctx.KVStore(k.StoreKey)
+
+				require.NoError(t, k.SaveDocument(ctx, tt.document))
```
Comment (gsora) [inline x/docs/internal/keeper/keeper_test.go:572]: If you must SaveDocument in multiple tests, try getting this line outside of various if/else cases: the less redundant the better.

File: x/docs/internal/keeper/keeper_test.go (hunk 2 above)
Comment (LucaArietti) [reply x/docs/internal/keeper/keeper_test.go:526]: empty set true means emptylist in testing

Comment (gsora) [reply x/docs/internal/keeper/keeper_test.go:526]: Uhm, instead of grouping most of the logic in separate if/else statement, I believe you can extract a common flow for all test cases and use small if's to differentiate various edge cases (list must be empty at the beginning, uuid must be different...).

File: x/docs/internal/keeper/keeper_test.go (hunk 3 above)
Comment (LucaArietti) [reply x/docs/internal/keeper/keeper_test.go:528]: In "existing receipt different Uuid" case (the old function TestKeeper_SaveDocumentReceipt_ExistingReceipt_DifferentUuid) oldReceipt and newReceipt are used
